pkg/util/validation/errors.go

```diff
@@ -210,6 +210,64 @@ func newExemplarMaxLabelLengthError(seriesLabels []mimirpb.LabelAdapter, exempla
 	}
 }
 
+type metadataMetricNameMissingError struct{}
+
+func newMetadataMetricNameMissingError() ValidationError {
+	return &metadataMetricNameMissingError{}
+}
+
+func (e *metadataMetricNameMissingError) Error() string {
+	return globalerror.MetricMetadataMissingMetricName.Message("received a metric metadata with no metric name")
+}
+
+// metadataValidationError is a ValidationError implementation suitable for metadata validation errors.
+type metadataValidationError struct {
+	message    string
+	cause      string
+	metricName string
+}
+
+func (e *metadataValidationError) Error() string {
+	return fmt.Sprintf(e.message, e.cause, e.metricName)
+}
+
+var metadataMetricNameTooLongMsgFormat = globalerror.MetricMetadataMetricNameTooLong.MessageWithLimitConfig(
+	maxMetadataLengthFlag,
+	// When formatting this error the "cause" will always be an empty string.
+	"received a metric metadata whose metric name length exceeds the limit,%s metric name: '%.200s'")
```

Comment:
```suggestion
	// When formatting this error the "cause" will always be an empty string.
	"received a metric metadata whose metric name length exceeds the limit, metric name: '%.200[2]s'")
```

https://go.dev/play/p/x5AYoEGF1t4